services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/RouteOptions.java

```diff
@@ -1104,7 +1104,7 @@ public URL toUrl(@NonNull String accessToken) {
         decodedUrl.getQuery(),
         decodedUrl.getRef()
       );
-      return new URL(encodedUri.toASCIIString());
+      return new URL(encodedUri.toASCIIString().replace("%20&%20", "%20%26%20"));
```

Comment:
This will only work for strings like "Me & you" but if the name is "My &cafe&" we would still crash.

Could you add a comment in code that this is just a specific workaround until we find a better solution to fix the encoding in general? I don't think we should close https://github.com/mapbox/mapbox-java/issues/1409 just yet.

Reply:
done